cpp/src/join/join_common_utils.cuh

```diff
@@ -281,12 +281,21 @@ __inline__ __device__ void add_pair_to_cache(size_type const first,
                                              size_type* joined_shared_r)
 {
   size_type my_current_idx{atomicAdd(current_idx_shared + warp_id, size_type(1))};
-
   // its guaranteed to fit into the shared cache
   joined_shared_l[my_current_idx] = first;
   joined_shared_r[my_current_idx] = second;
 }
 
+__inline__ __device__ void add_left_to_cache(size_type const first,
```

Comment:
```suggestion
__device__ void add_left_to_cache(size_type const first,
```
The compiler is smart enough to inline whenever possible. `__forceinline__` may bring visible speedups sometimes but that's really rare.

Reply:
Did you want to leave this function, and only remove `flush_left_cache`?